x-pack/plugins/cases/public/components/create/utils.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import type { CasePostRequest } from '../../../common';
+import { GENERAL_CASES_OWNER } from '../../../common';
+import type { ActionConnector } from '../../../common/types/domain';
+import { CaseSeverity } from '../../../common/types/domain';
+import type { CasesConfigurationUI } from '../../containers/types';
+import { normalizeActionConnector, getNoneConnector } from '../configure_cases/utils';
+import {
+  customFieldsFormDeserializer,
+  customFieldsFormSerializer,
+  getConnectorById,
+  getConnectorsFormSerializer,
+} from '../utils';
+import type { CreateCaseFormSchema } from './schema';
+
+type GetInitialCaseValueArgs = Partial<Omit<CasePostRequest, 'owner'>> &
+  Pick<CasePostRequest, 'owner'>;
+
+export const getInitialCaseValue = ({
+  owner,
+  connector,
+  ...restFields
+}: GetInitialCaseValueArgs): CasePostRequest => ({
+  title: '',
+  assignees: [],
+  tags: [],
+  category: undefined,
+  severity: CaseSeverity.LOW as const,
+  description: '',
+  settings: { syncAlerts: true },
+  customFields: [],
+  ...restFields,
+  connector: connector ?? getNoneConnector(),
+  owner,
+});
+
+const trimUserFormData = (
+  userFormData: Omit<CreateCaseFormSchema, 'connectorId' | 'fields' | 'syncAlerts' | 'customFields'>
+) => {
+  let formData = {
+    ...userFormData,
+    title: userFormData.title.trim(),
+    description: userFormData.description.trim(),
+  };
+
+  if (userFormData.category) {
+    formData = { ...formData, category: userFormData.category.trim() };
+  }
+
+  if (userFormData.tags) {
+    formData = { ...formData, tags: userFormData.tags.map((tag: string) => tag.trim()) };
+  }
+
+  return formData;
+};
+
+export const createFormDeserializer = (data: CasePostRequest): CreateCaseFormSchema => {
+  if (data == null || isEmpty(data)) {
+    const { connector, settings, customFields, ...restData } = getInitialCaseValue({
+      owner: data.owner,
```

Comment:
`data.owner` will never have a value because of the `if` above, what should we have here as default?

Reply:
Thanks! I removed the `if` check as `data` is always available.